src/coiio/imageio_defines.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <OpenImageIO/export.h>
+
+typedef int64_t stride_t;
+typedef uint64_t imagesize_t;
+
+#define AUTOSTRIDE 0x8000000000000000L  // int64_t min
```

Comment:
Does modern C have the equivalent of `const size_t AutoStride = ...` ?

I'm worried about `#define`ing a generic term like AUTOSTRIDE. Or is that typical of C these days?

Reply:
I don't think so I'm afraid. It's either #define or doing `extern stride_t AUTOSTRIDE;` and defining it in one (and only one) of the cpp files. Either way we might want to use OIIO_AUTOSTRIDE to guard against possible name clashes.

Reply:
If it's a macro, then let's prefix with OIIO_ (I just hate when other headers pollute the preprocessor symbols). If it's an extern, then let's use the same capitalization as we used in C++: AutoStride.

I think I'm preferring the extern solution, but I don't feel especially strongly about it.

Reply:
I still feel like we ought to prepend a namespace there to be hygenic since it's essentially a global. The same goes for enums and their variants (which should also really have the enum name embedded too). So we have a few options:

1. Shouty case. This _feels_ the most c-like to me, but maybe that's because I've been spending a lot of time in CUDA and OptiX recently and this is what they do.

```
extern stride_t OIIO_AUTOSTRIDE;      //< static constexpr stride_t AutoStride;
extern OIIO_TypeDesc OIIO_TYPE_STRING; //< static constexpr TypeDesc TypeString;

enum OIIO_OpenMode {
    OIIO_OPENMODE_CREATE,
    OIIO_OPENMODE_APPENDSUBIMAGE,
    OIIO_OPENMODE_APPENDMIPLEVEL,
};
```

2. Pascal case. This is most similar to what's in C++ and is basically just removing the colons from the C++ name.
```
extern stride_t OIIOAutoStride;
extern OIIOTypeDesc OIIOTypeString;

enum OIIOOpenMode {
    OIIOOpenModeCreate,
    OIIOOpenModeAppendSubImage,
    OIIOOpenModeAppendMipLevel,
};
```

3. Pascalsnake case. Same as 2 but replacing the colons with underscores:
```
extern stride_t OIIO_AutoStride;
extern OIIO_TypeDesc OIIO_TypeString;

enum OIIO_OpenMode {
    OIIO_OpenMode_Create,
    OIIO_OpenMode_AppendSubImage,
    OIIO_OpenMode_AppendMipLevel,
};
```

Reply:
>  3. Pascalsnake case. Same as 2 but replacing the colons with underscores:

This one is my preference, I think it's the easiest to read and also the easiest to mentally infer the correct C names from C++ and vice versa.